### PR TITLE
feat: show top banner

### LINKED ISSUE #13667: [CAL-3125] show top banner for broken integrations
when we show this:

image

we should also show the top banner and link to the app that needs to be reinstalled

CleanShot 2024-02-13 at 22 24 52@2x

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3125](https://linear.app/calcom/issue/CAL-3125/show-top-banner-for-broken-integrations)</sub>

### PR DESCRIPTION
**Pull request by VenkateshSircilla123**
## What does this PR do?

This will add  top banner for broken integration and when you click on goto app link it will takes you to app details page

Fixes #13667
/claim #13667

Screenshot from 2024-02-16 11-39-03

[Screencast from 16-02-24 11:35:21 AM IST.webm](https://github.com/calcom/cal.com/assets/113850972/e5f8ac7a-6c4b-465e-b66d-64e1ed0afd05)

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Go to apps
- Go to installed apps
- click conferencing
- You will get notified at the top banner if you required to reinstall the app

## Mandatory Tasks

- [ X ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/components/AppListCard.tsx [hunk 1]
```diff
@@ -85,52 +86,67 @@ export default function AppListCard(props: AppListCardProps) {
...
+    <>
+      {invalidCredential && (
+        <div className="fixed left-0 top-0 mb-3 w-[100vw]">
+          <TopBanner
+            text={`Oh no! Looks like permission expired or was revoked. Please reinstall again. `}
+            variant="warning"
+            actions={
+              <a href={`/apps/${slug}`} className="border-b border-b-black">
+                Go to App
+              </a>
```
Comment (ChaudharyPradip) [inline apps/web/components/AppListCard.tsx:98]: Don't we use a Link tag for the navigation?

Comment (VenkateshSircilla123) [reply apps/web/components/AppListCard.tsx:98]: > Don't we use a Link tag for the navigation?


yes we can.

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
@VenkateshSircilla123 you don't have to create Top banner in this file. check packages/features/shell/Shell.tsx file. You can check banner for calendarCredentialBanner

File: apps/web/components/AppListCard.tsx [hunk 2]
```diff
@@ -85,52 +87,67 @@ export default function AppListCard(props: AppListCardProps) {
...
-        {logo ? (
-          <img
-            className={classNames(logo.includes("-dark") && "dark:invert", "h-10 w-10")}
-            src={logo}
-            alt={`${title} logo`}
+    <>
+      {invalidCredential && (
+        <div className="fixed left-0 top-0 mb-3 w-[100vw]">
+          <TopBanner
+            text={`Oh no! Looks like permission expired or was revoked. Please reinstall again. `}
```
Comment (Udit-takkar) [inline apps/web/components/AppListCard.tsx:94]: you also have to use i18n

**Pull request closed without merging**